### PR TITLE
`Dependency` New Resource azurerm_healthcare_fhir_service

### PR DESCRIPTION
**Pull request by xiaxyi**
Adding FHIR service (Fast Healthcare Interoperability Resources) for the azure healthcare api2.0

The healthcare industry is rapidly transforming health data to the emerging standard of [FHIR](https://hl7.org/fhir)

FHIR service in Azure Health Data Services (hereby called the FHIR service) enables rapid exchange of data through Fast Healthcare Interoperability Resources (FHIR®) APIs, backed by a managed Platform-as-a Service (PaaS) offering in the cloud.

Dependency of pr #15759 
acc tests:
```
--- PASS: TestAccHealthCareFhirServiceDataSource_basic (903.25s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/healthcare    903.940s

--- PASS: TestAccHealthcareApiFhirService_basic (875.40s)
--- PASS: TestAccHealthcareApiFhirService_requiresImport (923.26s)
--- PASS: TestAccHealthcareApiFhirService_complete (978.85s)
--- PASS: TestAccHealthcareApiFhirService_updateIdentity (1177.58s)
--- PASS: TestAccHealthcareApiFhirService_updateAcrLoginServer (2527.39s)
--- PASS: TestAccHealthcareApiFhirService_updateCors (2618.35s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/healthcare    2619.134s
```

### CODE REVIEW

**Comment by JameySteinmann** [comment]
Thank you for putting in this work. I too am looking forward to using this resource as soon as it's available.

File: internal/services/healthcare/healthcare_fhir_resource.go [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+						"allowed_origins": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							MaxItems: 64,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"allowed_headers": {
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource.go:134]: could we be consistent with spacing here?
```suggestion
							},
						},

						"allowed_headers": {
```

File: internal/services/healthcare/healthcare_fhir_resource.go [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+									"GET",
+									"HEAD",
+									"MERGE",
+									"POST",
+									"OPTIONS",
+									"PUT",
+								}, false),
+							},
+						},
+						"max_age_in_seconds": {
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource.go:161]: etc

File: internal/services/healthcare/healthcare_fhir_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+  workspace_id        = azurerm_healthcare_workspace.test.id
+  kind                = "fhir-R4"
+  authentication {
+    authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47"
+    audience  = "https://acctestfhir.fhir.azurehealthcareapis.com"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+  depends_on = [azurerm_healthcare_workspace.test]
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource_test.go:190]: could we format this terraform betteR?
```suggestion
  kind                = "fhir-R4"
  
  authentication {
    authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47"
    audience  = "https://acctestfhir.fhir.azurehealthcareapis.com"
  }
  
  identity {
    type = "SystemAssigned"
  }
  
  depends_on = [azurerm_healthcare_workspace.test]
```

Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource_test.go:190]: how come we need a depends on here? this is often a sign of poor design

File: internal/services/healthcare/healthcare_fhir_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+  acr_login_servers = []
+  cors_configuration {
+    allowed_origins    = ["https://acctest.com:123"]
+    allowed_headers    = ["*"]
+    allowed_methods    = ["GET", "DELETE"]
+    max_age_in_seconds = 0
+    allow_credentials  = false
+  }
+  export_storage_account_name = azurerm_storage_account.test.name
+  depends_on                  = [azurerm_healthcare_workspace.test, azurerm_storage_account.test]
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource_test.go:363]: please fix all tf code

File: website/docs/r/healthcare_fhir_service.html.markdown [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+* `type` - (Required) The type of identity used for the Healthcare FHIR service. Possible values are `SystemAssigned`.
+
+---
+A `cors_configuration` block supports the following:  
+
+* `allowed_origins` - (Required) A set of origins to be allowed via CORS.
+* `allowed_headers` - (Required) A set of headers to be allowed via CORS.
+* `allowed_methods` - (Required) The methods to be allowed via CORS.
+* `max_age_in_seconds` - (Required) The max age to be allowed via CORS.
+* `allow_credentials` - (Boolean) If credentials are allowed via CORS.
```
Comment (katbyte) [inline website/docs/r/healthcare_fhir_service.html.markdown:88]: could we make this
```suggestion
* `credentials_allowed` - (Boolean) If credentials are allowed via CORS.
```

Comment (katbyte) [reply website/docs/r/healthcare_fhir_service.html.markdown:88]: also we do not add boolean here, this should be optional or required
```suggestion
* `allow_credentials` - (optional|required) If credentials are allowed via CORS.
```

File: website/docs/r/healthcare_fhir_service.html.markdown [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+* `access_policy_object_ids` - (Optional) A list of the access policies of the service instance.
+
+* `cors_configuration` - (Optional) A `cors_configuration` block as defined below.
+
+* `acr_login_servers` - - (Optional) A list of azure container registry settings used for convert data operation of the service instance.
+
+* `authentication` - (Required) An `authentication` block as defined below.
+
+* `export_storage_account_name` - (Optional) specifies the name of the export storage account which accepts the operation configuration information
```
Comment (katbyte) [inline website/docs/r/healthcare_fhir_service.html.markdown:72]: grammer, caps, punciation
```suggestion
* `export_storage_account_name` - (Optional) Specifies the name of the storage account which the operation configuration information is exported to.
```

please fix all documentation grammer, caps and punctuation.

also this property is unclear, maybe it should be `configuration_export_storage_account_name`

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/healthcare_fhir_service.html.markdown (hunk 5 above)
Comment (xiaxyi) [reply website/docs/r/healthcare_fhir_service.html.markdown]: I used the same property definition from the existing resource [healthcare_service](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/healthcare_service), should I change that resource too?

**Comment by xiaxyi** [comment]
@katbyte Thanks for the comments, updated the code

File: website/docs/r/healthcare_fhir_service.html.markdown (hunk 5 above)
Comment (katbyte) [reply website/docs/r/healthcare_fhir_service.html.markdown]: yes please

File: internal/services/healthcare/healthcare_fhir_resource.go [hunk 7]
```diff
@@ -0,0 +1,641 @@
...
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"identity": commonschema.SystemAssignedIdentityOptional(),
+
+			// can't use the registry ID due to the ID cannot be obtained when setting the property in state file
+			"acr_login_servers": {
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource.go:112]: what is this property? name? url? id? can we put what the user hsould supply in the name?
```suggestion
			"acr_login_server_urls": {
```

File: internal/services/healthcare/healthcare_fhir_resource.go [hunk 8]
```diff
@@ -0,0 +1,641 @@
...
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+
+			"cors_configuration": {
```
Comment (katbyte) [inline internal/services/healthcare/healthcare_fhir_resource.go:121]: configuration is redundant
```suggestion
			"cors": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/healthcare/healthcare_fhir_resource.go (hunk 8 above)
Comment (xiaxyi) [reply internal/services/healthcare/healthcare_fhir_resource.go]: updated, will change existing healthcare_service_resource.go as well

**Review by katbyte — APPROVED** [review]
LGTM 🦋

**Pull request merged**